test/e2e/e2e_test.go

```diff
@@ -222,6 +224,92 @@ func TestE2EFlow(t *testing.T) {
 		VerifyIncreaseOfUserAccountCount(t, originalToolchainStatus, currentToolchainStatus, targetedJohnMur.Spec.UserAccounts[0].TargetCluster, 1)
 	})
 
+	t.Run("verify userAccount is not deleted if namespace is not deleted", func(t *testing.T) {
+		laraUserName := "laracroft"
+		userNamespace := "laracroft-dev"
+		podName := "test-useraccount-delete-1"
+		laraSignUp := CreateAndApproveSignup(t, hostAwait, laraUserName, memberAwait.ClusterName)
+		VerifyResourcesProvisionedForSignup(t, hostAwait, laraSignUp, "base", memberAwait)
+
+		// when
+		//Create a pod with finalizer in user's namespace, which will block the deletion of namespace.
+		memberPod := corev1.Pod{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Pod",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      podName,
+				Namespace: userNamespace,
+				Finalizers: []string{
+					"kubernetes",
+				},
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{{
+					Name:  "test",
+					Image: "busybox",
+				}},
+			},
+		}
+		err = memberAwait.Client.Create(context.TODO(), &memberPod)
+		require.NoError(t, err)
+
+		// confirm pod created
+		pod := &corev1.Pod{}
+		err = memberAwait.Client.Get(context.TODO(), types.NamespacedName{Namespace: userNamespace, Name: podName}, pod)
+		require.NoError(t, err)
+
+		deletePolicy := metav1.DeletePropagationForeground
+		deleteOpts := &client.DeleteOptions{
+			PropagationPolicy: &deletePolicy,
+		}
+		// now delete userSignup but nothing should be deleted yet
+		err = hostAwait.Client.Delete(context.TODO(), laraSignUp, deleteOpts)
+		require.NoError(t, err)
+
+		err = memberAwait.WaitUntilNamespaceDeleted(laraUserName, "dev")
+		assert.Error(t, err, "laracroft-dev namespace got deleted")
+
+		err = memberAwait.WaitUntilNSTemplateSetDeleted(laraUserName)
+		assert.Error(t, err, "NSTemplateSet got deleted")
+
+		err = memberAwait.WaitUntilUserAccountDeleted(laraUserName)
+		assert.Error(t, err, "userAccount got deleted")
+
+		err = hostAwait.WaitUntilMasterUserRecordDeleted(laraUserName)
+		require.Error(t, err)
```

Comment:
In these calls, you wait for the whole timeout (which is 1 minute each) and check if the resource was deleted or not. You expect that all calls should return error - that the resource wasn't deleted during the 1 minute.
Since you do it 4 times, it probably waits 4 minutes - in that case you hit the threshold in MUR controller: https://github.com/codeready-toolchain/host-operator/blob/master/controllers/masteruserrecord/masteruserrecord_controller.go#L314-L316 which means that k8s starts scheduling and prolonging the periods between the reconciles. This is most likely the reason why it takes so long until the MUR is deleted when the UserSignup is successfully removed.

Instead of waiting 4 minutes, you should just wait for example 10 or max 20 seconds and check all resources alltogether. You can take a look at the other wait methods which check that a resource is not removed.

Reply:
This makes sense, trying it now.